src/engine/bank06.asm

```diff
@@ -307,23 +1333,1580 @@ Func_1a61f: ; 1a61f (6:661f)
+.asm_006_6937
+	ld a, [wd005]
+	ld c, a
+	ld a, h
+	bit D_LEFT_F, b
+	jr z, .asm_006_6974
+	; left
+	ld d, a
+	ld a, $06
+	cp l
+	ld a, d
+	jr nz, .asm_006_696b
+	push hl
+	push bc
+	push af
+	call GetCharacterInfoFromCursorPos
+	inc hl
+	inc hl
+	inc hl
+	inc hl
+	inc hl
+	ld a, [hl]
+	dec a
+	ld d, a
+	pop af
+	pop bc
+	pop hl
+	sub d
+	cp $ff
+	jr nz, .asm_006_6962
+	ld a, c
+	sub $02
+	jr .asm_006_69aa
+.asm_006_6962
+	cp $fe
+	jr nz, .asm_006_696b
+	ld a, c
+	sub $03
+	jr .asm_006_69aa
+.asm_006_696b
+	dec a
+	bit D_DOWN_F, a
+	jr z, .asm_006_69aa
+	ld a, c
+	dec a
+	jr .asm_006_69aa
+.asm_006_6974
+	bit D_RIGHT_F, b
+	jr z, .no_press
+	ld d, a
+	ld a, $06
+	cp l
+	ld a, d
+	jr nz, .asm_006_6990
+	push hl
+	push bc
+	push af
+	call GetCharacterInfoFromCursorPos
+	inc hl
+	inc hl
+	inc hl
+	inc hl
+	ld a, [hl]
+	dec a
+	ld d, a
+	pop af
+	pop bc
+	pop hl
+	add d
+.asm_006_6990
+	inc a
+	cp c
+	jr c, .asm_006_69aa
+	inc c
+	cp c
+	jr c, .asm_006_69a4
+	inc c
+	cp c
+	jr c, .asm_006_69a0
+	ld a, $02
+	jr .asm_006_69aa
+.asm_006_69a0
+	ld a, $01
+	jr .asm_006_69aa
+.asm_006_69a4
+	xor a
+	jr .asm_006_69aa
+.asm_006_69a7
+	ld l, a
+	jr .asm_006_69ab
+.asm_006_69aa
+	ld h, a
+.asm_006_69ab
+	push hl
+	call GetCharacterInfoFromCursorPos
+	inc hl
+	inc hl
+	inc hl
+	ld a, [wd009]
+	cp $02
+	jr nz, .asm_006_69bb
+	inc hl
+	inc hl
+.asm_006_69bb
+	ld d, [hl]
+	push de
+	call Func_006_6a07
+	pop de
+	pop hl
+	ld a, l
+	ld [wNamingScreenCursorY], a
+	ld a, h
+	ld [wNamingScreenCursorX], a
+	xor a
+	ld [wcea3], a
+	ld a, $06
+	cp d
+	jp z, NamingScreen_CheckButtonState
+	ld a, $01
+	ld [wcfe3], a
+.no_press
+	ldh a, [hKeysPressed]
+	and A_BUTTON | B_BUTTON
+	jr z, .asm_006_69ef
+	and A_BUTTON
+	jr nz, .asm_006_69e5
+	ld a, $ff
+.asm_006_69e5
+	call PlaySFXByA
+	push af
+	call Func_006_6a23
+	pop af
+	scf
+	ret
+.asm_006_69ef
+	ld a, [wcfe3]
+	or a
+	jr z, .asm_006_69f8
+	call PlaySFX
+.asm_006_69f8
+	ld hl, wcea3
+	ld a, [hl]
+	inc [hl]
+	and $0f
+	ret nz
+	ld a, [wceaa]
+	bit 4, [hl]
+	jr z, Func_006_6a07.asm_006_6a0a
+
+Func_006_6a07:
+	ld a, [wceab]
+.asm_006_6a0a
+	ld e, a
+	ld a, [wNamingScreenCursorX]
+	ld h, a
+	ld a, [wNamingScreenCursorY]
+	ld l, a
+	call GetCharacterInfoFromCursorPos
+	ld a, [hli]
+	ld c, a
+	ld b, [hl]
+	dec b
+	ld a, e
+	call Func_006_6a28
+	call WriteByteToBGMap0
+	or a
+	ret
+
+Func_006_6a23:
+	ld a, [wceaa]
+	jr Func_006_6a07.asm_006_6a0a
+
+Func_006_6a28:
+	push af
+	push bc
+	push de
+	push hl
+	push af
+	call ZeroObjectPositions
+	pop af
+	ld b, a
+	ld a, [wceab]
+	cp b
+	jr z, .asm_006_6a60
+	ld a, [wNamingScreenBufferLength]
+	srl a
+	ld d, a
+	ld a, [wNamingScreenBufferMaxLength]
+	srl a
+	ld e, a
+	ld a, d
+	cp e
+	jr nz, .asm_006_6a49
+	dec a
+.asm_006_6a49
+	ld hl, wNamingScreenNamePosition
+	add [hl]
+	ld d, a
+	ld h, $08
+	ld l, d
+	call HtimesL
+	ld a, l
+	add $08
+	ld d, a
+	ld e, $18
+	ld bc, $0000
+	call SetOneObjectAttributes
+.asm_006_6a60
+	pop hl
+	pop de
+	pop bc
+	pop af
+	ret
+
+SetVram0xFF:
```

Comment:
This is actually simply drawing ( in tile $00 of v0Tiles0) the blinking black square that is used in the name input in the position where the next inputted character is going to be displayed (don't know how to name that thing..., text cursor?)